app/rdf/queries.ts

```diff
@@ -218,19 +218,36 @@ export const getCubeDimensionValues = async ({
 }): Promise<DimensionValue[]> => {
   const { dimension, cube, locale, data } = rdimension;
 
-  if (
-    typeof dimension.minInclusive !== "undefined" &&
-    typeof dimension.maxInclusive !== "undefined" &&
-    data.dataKind !== "Time" &&
-    data.scaleType !== "Ordinal"
-  ) {
-    const min = parseObservationValue({ value: dimension.minInclusive }) ?? 0;
-    const max = parseObservationValue({ value: dimension.maxInclusive }) ?? 0;
-
-    return [
-      { value: min, label: `${min}` },
-      { value: max, label: `${max}` },
-    ];
+  if (data.dataKind !== "Time" && data.scaleType !== "Ordinal") {
+    if (
+      typeof dimension.minInclusive !== "undefined" &&
+      typeof dimension.maxInclusive !== "undefined"
+    ) {
+      const min = parseObservationValue({ value: dimension.minInclusive }) ?? 0;
+      const max = parseObservationValue({ value: dimension.maxInclusive }) ?? 0;
+
+      return [
+        { value: min, label: `${min}` },
+        { value: max, label: `${max}` },
+      ];
+    }
+
+    const firstPointer = dimension.out(ns.sh.or).out(ns.rdf.first);
```

Comment:
@bprusinowski just a first quick question:
would this also retrieve the second or third element in a list behind sh:or?

asking because I didn't see the part
 `rdf:rest*/rdf:first`

Reply:
@Rdataflow thanks for taking a look at this. I am not sure how to properly test this, but you're right that right now we only check for the first element in the list. I think I fixed the issue now by also considering `ns.rdf.rest` predicate (I am not sure if this will catch remaining elements, but the results for [this NFI dataset](https://test.visualize.admin.ch/en/create/new?cube=https://environment.ld.admin.ch/foen/nfi/nfi_C-2171/cube/2023-6) are the same in both approaches).

Reply:
@bprusinowski FYI rdf:List behaves like this ![image](https://github.com/visualize-admin/visualization-tool/assets/26090254/6a5a0b12-c3ad-44ae-bff9-007e2832abb0)

thus the goal it to retrieve
- `sh:or`
   - then step over **0, 1 or even many** `rdf:rest`\*
     - to get to the `rdf:first`
       - finally containing the relevant sh:minInclusive / sh:maxInclusive

I'm not sure about the effect of the array in this piece of code though / tom+ might know best
```
      .out(ns.sh.or)
      .out([ns.rdf.first, ns.rdf.rest]);
```

Reply:
I see, and is it guaranteed that there is only one min/max value across the whole list (so we can stop traversal as soon as there is a value found)? I am wondering why would there be several entries in this list, and assuming there are e.g. 10 entries, if 3rd has these values, what's in the remaining 7? I think there should only be one min/max values, attached directly either to dimension, or `sh:first`?

Reply:
@bprusinowski yes, there must be only 1 entry for minInclusive / maxInclusive property within the same List. go with this assumption. 👍 
- this needs to be guaranteed by the cube validator... see https;://cube.link/#validate-the-cube (still WIP as of now)

fyi: there might be other entries on datatype cube:Undefined and what more is or will be...

Reply:
LGTM 👍 
@kronmar can you give it a try?